src/System.IdentityModel.Tokens.Jwt/JwtSecurityTokenHandler.cs

```diff
@@ -674,79 +587,42 @@ public virtual ClaimsPrincipal ValidateToken(string securityToken, TokenValidati
         public override string WriteToken(SecurityToken token)
         {
             if (token == null)
-            {
-                LogHelper.Throw(string.Format(CultureInfo.InvariantCulture, LogMessages.IDX10000, GetType() + ": token"),typeof(ArgumentNullException), EventLevel.Verbose);
-            }
+                throw LogHelper.LogArgumentNullException("token");
 
             JwtSecurityToken jwt = token as JwtSecurityToken;
             if (jwt == null)
-            {
-                LogHelper.Throw(string.Format(CultureInfo.InvariantCulture, LogMessages.IDX10706, GetType(), typeof(JwtSecurityToken), token.GetType()), typeof(ArgumentNullException), EventLevel.Error);
-            }
+                throw LogHelper.LogException<ArgumentNullException>(LogMessages.IDX10706, new object[] { GetType(), typeof(JwtSecurityToken), token.GetType() });
```

Comment:
The error is meant to report that a SecurityToken other than a JwtSecurityToken was passed in. the Cast will return null, it token is not derived from JwtSecuritytoken.